src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -135,6 +139,9 @@ internal override MemberSemanticModel GetMemberModel(SyntaxNode node)
             return IsInTree(node) ? this : null;
         }
 
+        // We hide this property, as EnsureRootBoundForNullabilityIfNecessary can cause
```

Comment:
Do we actually use this property in a lot of places? Properties with big side effects make me nervous. If its only used in one or two places, consider not calling `EnsureRootBoundForNullabilityIfNecessary` leaving it to the caller instead, and just assert that _lazySnapshotmanager is populated.

Reply:
It's used in derived classes, so I'd prefer to keep the field private. I will change this to a method though, so the side effects aren't unexpected.

---
In reply to: [298375705](https://github.com/dotnet/roslyn/pull/36833#discussion_r298375705) [](ancestors = 298375705)